js/src/modals/ExportAccount/ExportAccount.js

```diff
@@ -0,0 +1,195 @@
+
+import FileSaver from 'file-saver';
+
+@observer
+class ExportAccount extends Component {
+  static contextTypes = {
+    api: PropTypes.object.isRequired
+  };
+
+  static propTypes = {
+    accounts: PropTypes.object.isRequired,
+    balances: PropTypes.object.isRequired,
+    newError: PropTypes.func.isRequired,
+    personalAccountsInfo: PropTypes.func.isRequired,
+    onClose: PropTypes.func.isRequired
+  };
+
+  static ExportStore = ExportStore;
+
+  componentDidMount () {
+
+  }
+
+  render () {
+    const { canExport } = ExportStore;
+
+    return (
+      <Portal
+        buttons={ [
+          <Button
+            icon={ <CancelIcon /> }
+            key='cancel'
+            label={
+              <FormattedMessage
+                id='export.accounts.button.cancel'
+                defaultMessage='Cancel'
+              />
+            }
+            onClick={ this.onClose }
+          />,
+          <Button
+            disabled={ !canExport }
+            icon={ <CheckIcon /> }
+            key='execute'
+            label={
+              <FormattedMessage
+                id='export.accounts.button.export'
+                defaultMessage='Export'
+              />
+            }
+            onClick={ this.onExport }
+          />
+        ] }
+        onClose={ this.onClose }
+        open
+        title={
+          <FormattedMessage
+            id='export.accounts.title'
+            defaultMessage='Export an Account'
+          />
+        }
+      >
+        { this.renderList() }
+      </Portal>
+    );
+  }
+
+  renderList () {
+    let { accounts } = this.props;
+
+    accounts = Object
+      .keys(accounts)
+      .map((address) => accounts[address]);
+
+    return (
+      <SelectionList
+        isChecked={ this.isSelected }
+        items={ accounts }
+        noStretch
+        onSelectClick={ this.onSelect }
+        renderItem={ this.renderAccount }
+      />
+    );
+  }
+
+  renderAccount = (account) => {
+    const { balances } = this.props;
+    const balance = balances[account.address];
+
+    return (
+      <AccountCard
+        account={ account }
+        balance={ balance }
+        password
+        store={ ExportStore }
+      />
+    );
+  }
+
+  isSelected = (account) => {
+    const { selectedAccounts } = ExportStore;
+
+    return selectedAccounts[account.address];
+  }
+
+  onSelect = (account) => {
+    ExportStore.toggleSelectedAccount(account.address);
+  }
+
+  onClose = () => {
+    this.props.onClose();
+  }
+
+  onExport = () => {
+    const { selectedAccounts, inputValue } = ExportStore;
+
+    Object
+      .keys(selectedAccounts)
+      .forEach((account) => {
+        if (selectedAccounts[account]) {
+          this.exportAddress(account, inputValue[account]);
+        }
+      });
+  }
+
+  exportAddress (account, password) {
+    const { parity } = this.context.api;
+    const { accounts, newError } = this.props;
+
+    parity.exportAccount(account, password)
```

Comment:
Would move the interaction to the store. (As well as onExport, i.e. all logic in the actual store as opposed to the component)